### PR TITLE
[FEATURE] cart patch

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: Cart/CartManager.php [hunk 1]
```diff
@@ -178,6 +178,25 @@ protected function updateCartApiEntity(ApiOrder $apiOrder)
     }
 
     /**
+     * updates the cart
+     *
+     * @param $data
```
Comment (christianbader) [inline Cart/CartManager.php]: type?

File: Cart/CartManager.php [hunk 2]
```diff
@@ -178,6 +178,25 @@ protected function updateCartApiEntity(ApiOrder $apiOrder)
...
+     * @param $user
+     * @param $locale
+     * @return null|Order
+     * @throws \Sulu\Bundle\Sales\OrderBundle\Order\Exception\OrderException
```
Comment (christianbader) [inline Cart/CartManager.php:187]: does it?

Comment (turbo-ele) [inline Cart/CartManager.php:187]: yes

File: Order/OrderManager.php [hunk 3]
```diff
@@ -142,7 +150,9 @@ public function save(
         $statusId = null,
         $flush = true
     ) {
-        if ($id) {
+        $isNewOrder = !$id;
```
Comment (christianbader) [inline Order/OrderManager.php:153]: ?

File: Cart/CartManager.php (hunk 1 above)
Comment (turbo-ele) [inline Cart/CartManager.php]: array

**Comment by christianbader** [comment]
else ack

File: Order/OrderManager.php (hunk 3 above)
Comment (turbo-ele) [inline Order/OrderManager.php:153]: if id is not passed, the order has to be new

**Pull request merged**